### ISSUE TITLE
`git-town branch` does not run during unfinished `sync`

### ISSUE THREAD

**Issue by sergej-koscejev** [git-town/git-town#4108]
### Steps to reproduce

When there is an unfinished `git-town sync` and I run `git-town branch`, it asks me what to do with the unfinished sync. But I want to just view the branch type because in this specific case I'm worried that I perhaps should have marked the branch as observed and git-town is misbehaving.

I think running `git-town branch` should be allowed even during unfinished sync.

### Command output

```shell
$ git-town branch --verbose
Git Town 16.4.1

(verbose) git version
git version 2.46.2

(verbose) git rev-parse --show-toplevel
/Users/skoscejev/Projects/foo

(verbose) git config -lz --includes --global
...omitted as likely irrelevant...

(verbose) git config -lz --includes --local
...omitted as likely irrelevant...

(verbose) git status --long --ignore-submodules
interactive rebase in progress; onto 4b59fe84786
Last command done (1 command done):
   pick afe8850c1d5 Remove quux
Next commands to do (5 remaining commands):
   pick b8acb57ef57 Remove foo
   pick e88cd9b97b0 Make bar lazy
  (use "git rebase --edit-todo" to view and edit)
You are currently rebasing branch 'bar' on '4b59fe84786'.
  (fix conflicts and then run "git rebase --continue")
  (use "git rebase --skip" to skip this patch)
  (use "git rebase --abort" to check out the original branch)

Unmerged paths:
  (use "git restore --staged <file>..." to unstage)
  (use "git add <file>..." to mark resolution)
	both modified:   foo/somefile
	both modified:   foo/anotherfile

no changes added to commit (use "git add" and/or "git commit -a")

unfinished Git Town command

You have an unfinished "sync" command
that ended on the "bar" branch
3 minutes ago. Please choose how to proceed.


0 > Quit without running anything
1   Continue the "sync" command after having resolved conflicts
2   Skip the current branch and continue the "sync" command on the next branch
3   Undo the previous "sync" command
4   Discard the unfinished state and run the new command
```

**Comment by kevgo** [comment]
Yes, the current thinking is that the previous Git Town command should be finished before running the next Git Town command. However, you make a good point that `git town branch` should be an exception. Probably also `git town repo` and `git town status`.

**Comment by sergej-koscejev** [comment]
`git town status` already works like that (it reports that there was an unfinished sync).

**Comment by kevgo** [comment]
This is now shipped in [Git Town 16.5](https://github.com/git-town/git-town/releases/tag/v16.5.0).
